core-build/gulp-core-build/src/IBuildConfig.ts

```diff
@@ -11,6 +11,14 @@ import { IExecutable } from './IExecutable';
  * @public
  */
 export interface IBuildConfig {
+  /**
+   * The maximum amount of time the build can run before being terminated.
+   * Specified in milliseconds. By default, there is no timeout.
+   *
+   * If set to zero (0), the build will never time out.
+   */
+  maxBuildTime: number;
```

Comment:
Any specific reason to not have it option similar to IExecutable?

Reply:
The one in IExecutable falls back to this one, so one of them must be defined.